packages/x-data-grid-premium/src/components/promptControl/GridToolbarPromptControl.tsx

```diff
@@ -0,0 +1,272 @@
+import * as React from 'react';
+import { styled } from '@mui/material/styles';
+import composeClasses from '@mui/utils/composeClasses';
+import useEventCallback from '@mui/utils/useEventCallback';
+import {
+  getDataGridUtilityClass,
+  GRID_CHECKBOX_SELECTION_FIELD,
+  gridColumnLookupSelector,
+  GridLogicOperator,
+  GridSingleSelectColDef,
+} from '@mui/x-data-grid';
+import { getValueOptions, getVisibleRows } from '@mui/x-data-grid/internals';
+import InputAdornment from '@mui/material/InputAdornment';
+import { DataGridPremiumProcessedProps } from '../../models/dataGridPremiumProps';
+import { GridApiPremium } from '../../models/gridApiPremium';
+import { useGridApiContext } from '../../hooks/utils/useGridApiContext';
+import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
+import { PromptResponse } from '../../hooks/features/promptControl/types';
+import { RecordButton } from './RecordButton';
+
+type OwnerState = DataGridPremiumProcessedProps;
+
+const useUtilityClasses = (ownerState: OwnerState, recording: boolean) => {
+  const { classes } = ownerState;
+
+  const slots = {
+    root: ['toolbarPromptControl', recording && 'toolbarPromptControl--recording'],
+    recordingIndicator: ['toolbarPromptControlRecordingIndicator'],
+    recordButton: ['toolbarPromptControlRecordButton'],
+    sendButton: ['toolbarPromptControlSendButton'],
+  };
+
+  return composeClasses(slots, getDataGridUtilityClass, classes);
+};
+
+const GridToolbarPromptControlRoot = styled('div', {
+  name: 'MuiDataGrid',
+  slot: 'ToolbarPromptControl',
+  overridesResolver: (_props, styles) => styles.toolbarPromptControl,
+})<{ ownerState: OwnerState }>({
+  flex: 1,
+  display: 'flex',
+  flexDirection: 'row',
+});
+
+function sampleData(
+  apiRef: React.MutableRefObject<GridApiPremium>,
+  rootProps: DataGridPremiumProcessedProps,
+) {
+  const columns = Object.values(gridColumnLookupSelector(apiRef));
+  const rows = rootProps.rows;
```

Comment:
This won't work with the data source, would it?
We can take row data from the row state instead.